### PR TITLE
IDENTITY-6529: Add unit tests for the identity-outbound-auth-oidc repo

### PR DESCRIPTION
**Pull request by GDRDABARERA**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -168,6 +168,31 @@
...
+            <dependency>
+                <groupId>org.powermock</groupId>
+                <artifactId>powermock-module-testng</artifactId>
+                <scope>test</scope>
+                <version>1.6.6</version>
+            </dependency>
+            <dependency>
+                <groupId>org.powermock</groupId>
+                <artifactId>powermock-api-mockito</artifactId>
+                <scope>test</scope>
```
Comment (mefarazath) [inline pom.xml]: Define a variable for version

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/resources/testng.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ ~ Copyright (c) 2016, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/resources/testng.xml]: Licence year should be 2017

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
+package org.wso2.carbon.identity.application.authenticator.oidc;
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: licence headers missing.

Comment (GDRDABARERA) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: done!

File: pom.xml (hunk 1 above)
Comment (GDRDABARERA) [reply pom.xml]: done!

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/TestUtils.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package org.wso2.carbon.identity.application.authenticator.oidc;
+
+import org.apache.commons.lang.StringUtils;
+import java.nio.file.Paths;
+
+/**
+ * Test util methods
+ */
+public class TestUtils {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/TestUtils.java]: Unless you have multiple test classes and want to use this util method in them better to move this to a private method in the test class itself.

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java [hunk 5]
```diff
@@ -0,0 +1,269 @@
...
+        };
+        jsonObject.put("stub", stub);
+        token = null;
+
+    }
+
+    @Test
+    public void testCanHandle() throws Exception {
+
+        mockRequest = mock(HttpServletRequest.class);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: why not move this to three separate test cases with some comments explaining what each test cases focuses on

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+        authenticatorProperties = new HashMap<>();
+        authenticatorProperties.put("callbackUrl","http://localhost:8080/playground2/oauth2client");
+        authenticatorProperties.put("commonAuthQueryParams", "Show error equals true.");
+        authenticatorProperties.put("UserInfoUrl","https://localhost:9443/oauth2/userinfo");
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_ID, "Q1nGJ8_lO4GGkOIESzEMJZBT39Ma");
+        authenticatorProperties.put(OIDCAuthenticatorConstants.CLIENT_SECRET, "I38YrW1VZabefzcyPkZurenPDV0a");
+        authenticatorProperties.put(OIDCAuthenticatorConstants.OAUTH2_TOKEN_URL, " https://localhost:9443/oauth2/token");
+
+        jsonObject = new HashMap<>();
+        Stub stub = new Stub() {
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: why do we have to do this?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/resources/testStream.xml [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+ ~
+ ~ Unless required by applicable law or agreed to in writing,
+ ~ software distributed under the License is distributed on an
+ ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ ~ KIND, either express or implied.  See the License for the
+ ~ specific language governing permissions and limitations
+ ~ under the License.
+ -->
+
+<note>
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/resources/testStream.xml]: Do we need this file?

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(null);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE)).thenReturn("active,OIDC");
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.LOGIN_TYPE)).thenReturn(null);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR)).thenReturn("Error Login.");
+        Assert.assertTrue(openIDConnectAuthenticator.canHandle(mockServletRequest), "Invalid can handle response for the request.");
+        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(null);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: we can simplify this right?
when `when(mockServletRequest.getParameter(anyString())).thenReturn(null);`

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java [hunk 9]
```diff
@@ -0,0 +1,350 @@
...
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR)).thenReturn("Error Login.");
+        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(null);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_PARAM_STATE)).thenReturn("active,OIDC");
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.LOGIN_TYPE)).thenReturn(null);
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_ERROR)).thenReturn("Error Login.");
+        Assert.assertNotNull(openIDConnectAuthenticator.getContextIdentifier(mockServletRequest), "Invalid context identifier.");
+
+        when(mockServletRequest.getParameter(OIDCAuthenticatorConstants.OAUTH2_GRANT_TYPE_CODE)).thenReturn(null);
```
Comment (mefarazath) [inline components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: Let's simplify this
`when(mockServletRequest.getParameter(anyString())).thenReturn(null);`

File: components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java (hunk 8 above)
Comment (GDRDABARERA) [reply components/org.wso2.carbon.identity.application.authenticator.oidc/src/test/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticatorTest.java]: done

**Pull request merged**